lib/PrintAsClang/PrintClangFunction.cpp

```diff
@@ -77,62 +77,75 @@ class CFunctionSignatureTypePrinter
         languageMode(languageMode), typeUseKind(typeUseKind) {}
 
   bool printIfKnownSimpleType(const TypeDecl *typeDecl,
-                              Optional<OptionalTypeKind> optionalKind) {
+                              Optional<OptionalTypeKind> optionalKind,
+                              bool isInOutParam) {
     auto knownTypeInfo = getKnownTypeInfo(typeDecl, typeMapping, languageMode);
     if (!knownTypeInfo)
       return false;
     os << knownTypeInfo->name;
-    if (knownTypeInfo->canBeNullable)
+    if (knownTypeInfo->canBeNullable) {
       printNullability(optionalKind);
+    } else if (isInOutParam) {
```

Comment:
I don't think this should be an `else if`, but rather an `if`. You can have a nullable `inout`, e.g. `inout UnsafeRawPointer`. Can you fix it and add a test to catch it?

Reply:
Sure